ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py

```diff
@@ -277,3 +277,24 @@ def mod(
     if ivy.is_array(where):
         ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
     return ret
+
+
+def fmod(
```

Comment:
To handle dtype and casting we have a decorator for that no need for these if statements.
You will need to add these decorators to the function
```suggestion
@handle_numpy_dtype
@to_ivy_arrays_and_back
@handle_numpy_casting
@from_zero_dim_arrays_to_scalar
def fmod(
```